feat(ui): add Token Estimator link to footer

HmbleCreator opened the pull request:
### Summary

This PR adds a "Token Estimator" link to the footer of the Gitingest web frontend. The link points to [https://gitingest.com/tokencount](https://gitingest.com/tokencount), allowing users to easily access a tool for estimating token counts in pasted text.

### Details

- **UI Change:**  
  - Added a new link labeled **Token Estimator** (with a counter icon) to the left column of the footer, alongside the Extension and Python package links.
  - The link opens in a new tab and is styled consistently with the other resource links.
  - No changes were made to the CLI or Python package.

- **Why:**  
  - Addresses and **closes #318** by making token estimation more visible and accessible to users.

- **How to test:**  
  1. Start the server (`cd src && uvicorn server.main:app --reload`).
  2. Open the app in your browser.
  3. Scroll to the footer and verify the "Token Estimator" link appears and opens [https://gitingest.com/tokencount](https://gitingest.com/tokencount) in a new tab.

### Screenshot
![Screenshot 2025-06-29 205833](https://github.com/user-attachments/assets/c1d07a4e-ed61-4826-9cb4-b3bc2c53d9f5)

---

**Closes #318**

HmbleCreator committed:
feat(ui): add Token Estimator link to footer

- Adds a "Token Estimator" link (with icon) to the left column of the site footer.
- Link points to https://gitingest.com/tokencount and opens in a new tab.
- Styled consistently with other resource links.
- No changes to CLI or Python package.

Closes #318

HmbleCreator commented:
> @HmbleCreator Thanks, But this is just the link, it seems to me that #318 would require to actually implement the backend route for this new function, do you think you can work on this?

Thanks for pointing that out! Yes, I’d be happy to implement the backend route for token estimation as part of #318. I can add an endpoint (e.g. /api/estimate-tokens) that accepts raw text, estimates token count using tiktoken or a similar utility, and returns the result. Let me know if you have a preferred response schema or want it namespaced differently—happy to get started.

HmbleCreator commented:
> > @HmbleCreator Thanks, But this is just the link, it seems to me that #318 would require to actually implement the backend route for this new function, do you think you can work on this?
> 
> Thanks for pointing that out! Yes, I’d be happy to implement the backend route for token estimation as part of #318. I can add an endpoint (e.g. /api/estimate-tokens) that accepts raw text, estimates the token count using tiktoken or a similar utility, and returns the result. Let me know if you have a preferred response schema or want it namespaced differently; I'd be happy to get started.

Are we only want to have support for OpenAI models or other models also, like open source models available via HF?

HmbleCreator committed:
feat: Add token estimator documentation page and simplify footer link text

- Add GET endpoint for /api/tokencount with interactive documentation
- Include API usage examples and interactive form for testing

HmbleCreator committed:
feat: Add token estimator documentation page and simplify footer link text

- Add GET endpoint for /api/tokencount with interactive documentation
- Include API usage examples and interactive form for testing

ix-56h commented on src/server/routers/index.py:
```diff
@@ -74,3 +122,125 @@ async def index_post(request: Request, form: Annotated[QueryForm, Depends(QueryF
+@router.get("/api/tokencount", response_class=HTMLResponse)
+async def api_token_count_docs():
+    """Serve a simple documentation page for the token count API."""
+    html = f'''
```
Create a jinja template for the documentation, refer to the other api docs route `/api` and his template `server/templates/api.jinja`.

HmbleCreator committed:
feat: use Jinja template for token estimator API docs

- Add tokencount_api.jinja for styled token estimator documentation
- Update /api/tokencount GET endpoint to render the new template
- Documentation now matches the look and feel of other API docs

ix-56h commented:
Seems good, i'll test it out in few hours.

I see that both logic and routing are defined and implemented within the entrypoint of the server, maybe move this to the "routers/" folder.

ix-56h commented on tests/test_server.py:
```diff
@@ -0,0 +1,23 @@
+client = TestClient(app, base_url="http://localhost")
+
+
+def test_tokencount_valid():
```
If you keep the `/api/tokencount` as it is right now, no changes have to be made here.

filipchristiansen committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main

HmbleCreator committed:
Refactor token count API and routing, enforce API-only for /api/tokencount, and backend cleanup

- Removed GET /api/tokencount route that rendered a Jinja template; /api/tokencount is now POST-only and returns JSON.
- Renamed and refactored tokencount_api.jinja to tokencount.jinja, using a select dropdown for model selection and matching git_form.jinja UX.
- Updated footer link to point to /tokencount (user-facing form) instead of /api/tokencount.
- Ensured /tokencount is the user-facing form (GET/POST, Jinja) and /api/tokencount is API-only (POST, JSON).
- Updated dependency management: added autotiktokenizer to requirements.txt and pyproject.toml, and added a guideline to CONTRIBUTING.md.
- Cleaned up unused imports and dead code in backend.

HmbleCreator committed:
Merge branch 'main' of https://github.com/HmbleCreator/gitingest

HmbleCreator commented:
Hi!
I wanted to ask—does the UI theme/design you use for Gitingest have a name or is it based on a particular design system? I liked the style; it’s clean, modern, and playful!
Additionally, I wanted to express my gratitude for your guidance and patience over the past two to three days. As a newbie, it was a great experience learning about FastAPI, backend/frontend separation, API best practices, and collaborative open-source workflows. I learned a great deal and truly appreciate the opportunity to contribute!
Thanks again!

filipchristiansen committed:
Merge branch 'main' into main

ix-56h commented:
> Hi! I wanted to ask—does the UI theme/design you use for Gitingest have a name or is it based on a particular design system?
Yes, neobrutalism, you can find it at [neobrutalism.com](https://www.neobrutalism.dev/).

> I wanted to express my gratitude for your guidance and patience over the past two to three days. As a newbie, it was a great experience learning about FastAPI, backend/frontend separation, API best practices, and collaborative open-source workflows. I learned a great deal and truly appreciate the opportunity to contribute! Thanks again!

You're welcome :)

filipchristiansen committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main

filipchristiansen committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'cyclotruc:main' into main

filipchristiansen committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main

ix-56h commented:
@HmbleCreator i'm sorry, we're late on this one.

We currently have some urgent topic and we don't know if we still want to add a dedicated tiktoken feature to the frontend app.

I'll put this in draft for now.

HmbleCreator committed:
Merge branch 'main' into main

HmbleCreator committed:
Merge branch 'main' into main
